src/plugins/intel_cpu/src/utils/custom_shape_inference/transpose.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transpose.hpp"
+#include "utils.hpp"
+#include "ie_ngraph_utils.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+using namespace InferenceEngine;
+TransposeShapeInfer::TransposeShapeInfer(const size_t& out_rank, const std::vector<size_t>& axes_vec)
+    : m_out_rank(out_rank), m_axes_vec(axes_vec), m_outputShape(out_rank, 1), m_needReverse(axes_vec.empty()) {}
+
+Result TransposeShapeInfer::infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) {
+    const VectorDims& shapeIn = input_shapes[0].get();
+    if (m_needReverse) {
+        for (size_t i = 0; i < m_out_rank; ++i) {
+            m_outputShape[i] = shapeIn[m_out_rank - 1 - i];
+        }
+    } else {
+        for (size_t i = 0; i < m_out_rank; ++i) {
+            m_outputShape[i] = shapeIn[m_axes_vec[i]];
+        }
+    }
+    return {{m_outputShape}, ShapeInferStatus::success};
+}
+ShapeInferPtr TransposeShapeInferFactory::makeShapeInfer() const {
+    if (const auto order = ov::as_type_ptr<const ov::op::v0::Constant>(m_op->get_input_node_shared_ptr(ov::op::v1::Transpose::ORDER))) {
+        const auto axes_vec = order->cast_vector<size_t>();
+        return std::make_shared<TransposeShapeInfer>(m_op->get_output_partial_shape(0).rank().get_length(), axes_vec);
+    } else {
+        return std::make_shared<TransposeDynShapeInfer>();
```

Comment:
Shouldn't we use ngraph version here?

Reply:
I am not sure,  please check [comment](https://github.com/openvinotoolkit/openvino/pull/15695#discussion_r1112075697)

Reply:
Ok, got it. Thanks.